### PR TITLE
add extension version

### PR DESCRIPTION
**Pull request by HarrisChu**
I write a k6 extension and build with xk6, but I cannot know the extension version via command line.

if extension module implement `VersionGetter` interface, could display via version command.

```
package k6plugin

import (
	"github.com/vesoft-inc/k6-plugin/pkg/aggcsv"
	"github.com/vesoft-inc/k6-plugin/pkg/nebulagraph"
	"go.k6.io/k6/js/modules"
	"go.k6.io/k6/output"
)

func init() {
	modules.Register("k6/x/nebulagraph", nebulagraph.NewNebulaGraph())
	output.RegisterExtension("aggcsv", func(p output.Params) (output.Output, error) {
		return aggcsv.New(p)
	})
}
```

```
> ./k6 version
k6 v0.40.0 ((devel), go1.18.5, linux/amd64)
extension k6/x/nebulagraph v1.0.2
```

### CODE REVIEW

**Comment by codebien** [comment]
Hey @HarrisChu,
thanks for your contribution :bow:

Please, check [this previous comment](https://github.com/grafana/k6/pull/2686#pullrequestreview-1113579243) where we shared our vision regarding this feature.

**Comment by HarrisChu** [comment]
> 

perfect for me.
so I should close this pr?

**Comment by codebien** [comment]
If you would like to contribute then you can keep this open and address it as a request change.

**Comment by HarrisChu** [comment]
I cannot test the code as `debug.ReadBuildInfo` not populated for test
just test it with my extension.

image

File: cmd/version.go
Comment (imiric) [inline cmd/version.go:22]: k6 supports both JS and output extensions. So we should make sure that both are shown in this output (same goes for the `k6 run` output).

You can test by building a binary with an output extension, e.g.:
```
XK6_K6_REPO=github.com/HarrisChu/k6 xk6 build master \
  --with github.com/vesoft-inc/k6-plugin \
  --with github.com/grafana/xk6-output-influxdb
```

It should be a matter of doing something similar as in `js/modules/modules.go` and calling `getPackageVersion(mod)` from [`output.RegisterExtension()`](https://github.com/grafana/k6/blob/e14f14d416b0af4e495e6f565a2a9787a73ade54/output/extensions.go#L27). You'll probably want to move `getPackageVersion()` somewhere where it's accessible by both (maybe `lib/extensions.go`?).

File: js/modules/modules.go
Comment (imiric) [inline js/modules/modules.go:21]: Would it make sense to reuse the `modules` map, and have its values be a small struct like:

```go
type module struct {
	mod     interface{}
	version string
}
```

This way `getPackageVersion()` would only return the version, as its name implies, and not also set (another) global map value.

This will complicate things slightly, as you'll need to change wherever `GetJSModules()` is called (or change `GetJSModules()` itself to continue to return just the modules without the version; though I think the first approach is slightly better), but it should avoid the repetition, and be a bit cleaner in the end. It would also avoid the need for a separate `GetJSModuleVersions()` function.

File: js/modules/modules.go
Comment (imiric) [inline js/modules/modules.go:43]:
```suggestion
func getModuleVersion(mod interface{}) {
```

I think this is slightly clearer, as we are getting the Go _module_ version after all (and the map is called `moduleVersions`).

**Review by imiric — CHANGES_REQUESTED** [review]
Thanks for this PR @HarrisChu! :bow:

We've had #1741 in our backlog for a long time, so it's great that it's finally being implemented. As you've found, it's an important feature for k6 extension maintainers.

I do have a couple of requirements before merging this. Namely making the `k6 run` output more pleasing, and also showing the versions of any output extensions. The others are suggestions that would be nice to have, but not blockers.

Since testing this is a bit tricky, we might want to have an E2E test instead. Currently we have an [`xk6` CI workflow](https://github.com/grafana/k6/blob/e14f14d416b0af4e495e6f565a2a9787a73ade54/.github/workflows/xk6.yml), so it might make sense to add an assertion or two about the actual output of both `k6 run` and `k6 version` there. Unless someone has a better idea to test this...

File: cmd/ui.go
Comment (HarrisChu) [reply cmd/ui.go:91]: ok

File: js/modules/modules.go
Comment (HarrisChu) [reply js/modules/modules.go:21]: looks good

File: js/modules/modules.go
Comment (HarrisChu) [reply js/modules/modules.go:43]: good catch.

btw, If a package has two modules, it just has one package in map.
how about display the module name? e.g.

```
 extensions: github.com/dgzlopes/k6-extension-notifications k6/x/ext1 v1.0.0
                     github.com/dgzlopes/k6-extension-notifications k6/x/ext2 v1.0.0
```

**Comment by imiric** [comment]
Hi @HarrisChu, are you available to finish the work on the pending items? We'd like to release this in the upcoming v0.42.0, so this should be merged in the next week or two.

If not, that's OK as well, and someone from our side would pick it up. You'd still get credit in the release notes, of course :slightly_smiling_face:

**Review by oleiade — CHANGES_REQUESTED** [review]
I align with the comments and recommendations of @imiric 🙏🏻 
Really good work though, looking forward to see those comments addressed 👍🏻

**Comment by imiric** [comment]
Closing this since it was superseded by #2805. Thanks again for your contribution @HarrisChu! :bow:

**Comment by HarrisChu** [comment]
> 

sorry, I had some emergencies...
glad it's already in v0.42, I would update our plugins.

**Comment by imiric** [comment]
@HarrisChu No problem.

It didn't make it in v0.42.0, but it will be released in v0.43.0 a few weeks from now.

**Pull request closed without merging**
